sql/parser/datum.go

```diff
@@ -751,8 +751,10 @@ func (d DInterval) IsMin() bool {
 
 // String implements the Datum interface.
 func (d DInterval) String() string {
-	// TODO(dan): Make this output postgres compatible for all values.
-	return time.Duration(d.Duration.Nanos).String()
+	if d.Months != 0 || d.Days != 0 {
+		return d.String()
```

Comment:
I'm surprised this works - you don't have to `d.Duration.String()` to keep this from being recursive?